android/src/main/java/com/brentvatne/react/ReactVideoView.java

```diff
@@ -203,11 +236,18 @@ public void setSrc(final String uriString, final String type, final boolean isNe
         src.putString(ReactVideoViewManager.PROP_SRC_URI, uriString);
         src.putString(ReactVideoViewManager.PROP_SRC_TYPE, type);
         src.putBoolean(ReactVideoViewManager.PROP_SRC_IS_NETWORK, isNetwork);
+        if(mMainVer>0) {
+            src.putInt(ReactVideoViewManager.PROP_SRC_MAINVER, mMainVer);
+            if(mPatchVer>0) {
+                src.putInt(ReactVideoViewManager.PROP_SRC_PATCHVER, mPatchVer);
+            }
+        }
         WritableMap event = Arguments.createMap();
         event.putMap(ReactVideoViewManager.PROP_SRC, src);
         mEventEmitter.receiveEvent(getId(), Events.EVENT_LOAD_START.toString(), event);
 
-        prepareAsync(this);
+        // not async to prevent random crashes on Android playback from local resource due to race conditions
+        prepare(this);
```

Comment:
I couldn't compile my project, received the following error message `error: unreported exception IOException; must be caught or declared to be thrown
        prepare(this);`

I changed it back to `prepareAsync(this)` and was able to complie